### DISCUSSION TITLE
Duckdb v.1.1.0 How to (conditionally) set variable based on platform (Windows vs Linux)

### DISCUSSION THREAD

**Discussion by jimdickson007** [duckdb/duckdb#13916] [Q&A] [answered]
I am aware of 
```
PRAGMA platform;
CALL pragma_platform();
```

Not aware how to use either of these to set variable 

Goal is to set a variable to a directory path - but this varies on Window / Linux

**Comment by carlopi** [comment]
This should work:
```
SELECT platform FROM pragma_platform();
```

**Reply by carlopi** [reply, marked as answer]
More complex example, using `v1.1.0` variables:

```
SET variable platform = (SELECT platform FROM pragma_platform());
SELECT getvariable('platform');
```

Courtesy of this PR https://github.com/duckdb/duckdb/pull/13084

**Comment by Maxxen** [comment]
The `parse_dirname` and `parse_dirpath` [functions](https://duckdb.org/docs/sql/functions/char.html#parse_dirnamepath-separator) might also be interersting
